packages/geo/src/tile.matrix.set.ts

```diff
@@ -131,4 +125,89 @@ export class TileMatrixSet {
         const ul = this.tileToPixels(tile.x, tile.y);
         return this.pixelsToSource(ul.x, ul.y, tile.z);
     }
+
+    /**
+     * Get the boundingBox for a `tile` in source units
+     */
+    public tileToSourceBounds(tile: Tile): Bounds {
+        const ul = this.tileToSource(tile);
+        const width = this.pixelScale(tile.z) * this.tileSize;
+        return new Bounds(ul.x, ul.y - width, width, width);
+    }
+
+    /**
+     * Iterate over the top level tiles that cover the extent of the `TileMatrixSet`
+     */
+    *topLevelTiles(): Generator<Tile, null, void> {
+        const z = 0;
+        const { matrixWidth, matrixHeight } = this.zooms[0];
+
+        for (let y = 0; y < matrixHeight; ++y) {
+            for (let x = 0; x < matrixWidth; ++x) {
+                yield { x, y, z };
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Iterate over the child (`z+1`) tiles that cover `tile`
+
+     * @param tile
+     */
+    *coverTile(tile?: Tile): Generator<Tile, null, void> {
+        if (tile == null) {
+            yield* this.topLevelTiles();
+            return null;
+        }
+        const z = tile.z + 1;
+        const { matrixWidth, matrixHeight } = this.zooms[tile.z];
+        const { matrixWidth: pw, matrixHeight: ph } = this.zooms[z];
+
+        const xfact = pw / matrixWidth;
+        const yfact = ph / matrixHeight;
+
+        const xs = Math.floor(tile.x * xfact);
+        const ys = Math.floor(tile.y * yfact);
+
+        const xe = Math.floor((tile.x + 1) * xfact);
+        const ye = Math.floor((tile.y + 1) * yfact);
+
+        for (let y = ys; y < ye; ++y) {
+            for (let x = xs; x < xe; ++x) {
+                yield { x, y, z };
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Make a name for a `tile`
+     */
+    public static tileToName(tile: Tile): string {
+        return `${tile.z}-${tile.x}-${tile.y}`;
```

Comment:
this is different to a lot of places where we have `{x}_{y}_z{z}`

do we want to prefix the keys `z${z}_x${x}_y${y}`

Reply:
Lets just move our convention to z-x-y